core/lens/lens_seq.ml

```diff
@@ -0,0 +1,41 @@
+type 'a t = unit -> 'a node
```

Comment:
Did you copy this module or base it directly on the OCaml module `Seq`? If so, you need to acknowledge the OCaml project in the header.

Reply:
Added a comment regarding this in both files. It is only based of the signatures and the comments.